### PR TITLE
Multi shotgun fix

### PR DESCRIPTION
**Pull request by MEEPofFaith**
Fixes a `ShootMulti` with a `ShootBarrel`/`ShootAlternate` source skipping barrels if the dest shoots fire multiple bullets.

Before:

https://user-images.githubusercontent.com/54301439/229329201-e55ba72d-8156-4d33-8fb2-aee2a97e8acf.mp4

After:

https://user-images.githubusercontent.com/54301439/229329206-182ecb10-bdc7-457c-8a3d-09d4d2d2f240.mp4

---
Additionally also implements multi recoil.

https://user-images.githubusercontent.com/54301439/229329218-f6537671-6ac6-4b4b-8e48-b04bb2b0a649.mp4

---

If your pull request is **not** translation or serverlist-related, read the list of requirements below and check each box:

- [X] I have read the [contribution guidelines](https://github.com/Anuken/Mindustry/blob/master/CONTRIBUTING.md).
- [X] I have ensured that my code compiles, if applicable.
- [X] I have ensured that any new features in this PR function correctly in-game, if applicable.

### CODE REVIEW

File: core/src/mindustry/world/blocks/defense/turrets/Turret.java [hunk 1]
```diff
@@ -210,8 +212,9 @@ public class TurretBuild extends ReloadTurretBuild implements ControlBlock{
         public Seq<AmmoEntry> ammo = new Seq<>();
         public int totalAmmo;
         public float curRecoil, heat, logicControlTime = -1;
+        public float[] curRecoils;
```
Comment (MEEPofFaith) [inline core/src/mindustry/world/blocks/defense/turrets/Turret.java:215]: Do I need `@Nullable` on this?

File: core/src/mindustry/entities/pattern/ShootPattern.java [hunk 2]
```diff
@@ -12,12 +12,17 @@ public class ShootPattern implements Cloneable{
     public float shotDelay = 0;
 
     /** Called on a single "trigger pull". This function should call the handler with any bullets that result. */
-    public void shoot(int totalShots, BulletHandler handler){
+    public void shoot(int totalShots, BulletHandler handler, Runnable barrelIncrementer){
```
Comment (Anuken) [inline core/src/mindustry/entities/pattern/ShootPattern.java:15]: should be `@Nullable` for documentation

File: core/src/mindustry/world/blocks/defense/turrets/Turret.java (hunk 1 above)
Comment (Anuken) [reply core/src/mindustry/world/blocks/defense/turrets/Turret.java:215]: if any variable can be null, yes

File: core/src/mindustry/entities/units/WeaponMount.java [hunk 3]
```diff
@@ -14,6 +14,8 @@
     public float rotation;
     /** weapon recoil */
     public float recoil;
+    /** weapon barrel recoil */
+    public float[] recoils;
```
Comment (Anuken) [inline core/src/mindustry/entities/units/WeaponMount.java:18]: `@Nullable`

**Comment by MEEPofFaith** [comment]
Oh boy @BlueWolf3682 is going to have a field day with this.

**Pull request merged**